roles/satellite-clone/files/disassociate_capsules.rb

```diff
@@ -11,42 +12,39 @@ def prepare_hammer_cmd(command)
 end
 
 def run_hammer_cmd(command)
-  command = prepare_hammer_cmd(command)
-  `#{command}`
+  `#{prepare_hammer_cmd(command)}`
 end
 
-def get_info_from_hammer(command, column=1)
-  bash_parse = " | grep -v \"Warning:\" | tail -n+2 | awk -F, {'print $#{column}'}"
-  run_hammer_cmd(command + bash_parse)
+def get_ids_from_hammer(command, search = nil)
+  JSON.parse(run_hammer_cmd("--output=json " + command + (search ? " --search '#{search}'" : "")).map do |obj|
+    obj["Id"]
+  end
 end
 
 def capsule_lce_args(action, capsule_id, env)
   "--csv capsule content #{action}-lifecycle-environment --id #{capsule_id} --lifecycle-environment-id #{env}"
 end
 
-external_capsules = []
-external_capsule_ids = get_info_from_hammer("--csv capsule list --search 'feature = \"Pulp Node\"'")
-if external_capsule_ids.empty?
-  STDOUT.puts "There are no external capsules to disassociate."
-else
-  external_capsule_ids.split("\n").each do |id|
-    lifecycle_environment = get_info_from_hammer("--csv capsule content lifecycle-environments --id #{id}").split("\n")
-    name = get_info_from_hammer("--csv capsule info --id #{id}", 2).chomp
-    external_capsules << {:id => id, :name => name, :lifecycle_environments => lifecycle_environment}
-  end
+external_capsule_lifecycle_environments = get_ids_from_hammer("capsule list", search = 'feature = "Pulp Node"').map do |id|
+  { capsule_id: id, lifecycle_environments: get_ids_from_hammer("capsule content lifecycle-environments --id #{id}") }
+end
 
-  reverse_commands = []
-  external_capsules.each do |capsule|
-    capsule[:lifecycle_environments].each do |env|
-      run_hammer_cmd(capsule_lce_args("remove", capsule[:id], env))
-      reverse_commands << prepare_hammer_cmd(capsule_lce_args("add", capsule[:id], env))
-    end
+reverse_commands = external_capsule_lifecycle_environments.map do |association|
+  association[:lifecycle_environments].map do |env|
+    run_hammer_cmd(capsule_lcs_args("remove", association[:capsule_id], env))
+    prepare_hammer_cmd(capsule_lce_args("add", association[:capsule_id], env))
   end
+end.flatten
 
-  STDOUT.puts "All Capsules are unassociated with any lifecycle environments. This is to avoid any syncing errors with your original Satellite " \
-              "and any interference with existing infrastructure. To reverse these changes, run the following commands," \
-              " making sure to replace the credentials with your own."
-  reverse_commands.each do |reverse|
-    STDOUT.puts reverse
-  end
+if reverse_commands.empty?
+  STDOUT.puts "There were no associated lifecycle environments to disassociate from external capsules."
+else
+  STDOUT.puts <<~MESSAGE
```

Comment:
this doesn't work on Ruby 2.0 (which we still have to support), as it doesn't know "squiggly heredoc" (2.3+ feature)

Reply:
Oof. In what cases do we still support Ruby 2.0?

Reply:
EL7

Reply:
Ah right, because this script uses system ruby. TY